backend/aspnet/MShare_ASP/Services/SpendingService.cs

```diff
@@ -178,23 +178,77 @@ public async Task UpdateSpending(long userId, SpendingUpdate spendingUpdate)
 
             currentSpending.Name = spendingUpdate.Name;
             currentSpending.MoneyOwed = spendingUpdate.MoneySpent;
+			Context.Debtors.RemoveRange(currentSpending.Debtors);
 
-            Context.Debtors.RemoveRange(currentSpending.Debtors);
+			var debtors = spendingUpdate.Debtors.Select(x => new DaoDebtor()
+			{
+				Spending = currentSpending,
+				DebtorUserId = x.DebtorId,
+				Debt = x.Debt
+			}).ToList();
 
-            var debtors = spendingUpdate.Debtors.Select(x => new DaoDebtor()
-            {
-                Spending = currentSpending,
-                DebtorUserId = x.DebtorId,
-                Debt = x.Debt
-            }).ToList();
+			currentSpending.Debtors = debtors.ToList();
 
-            currentSpending.Debtors = debtors.ToList();
+			await OptimizedService.OptimizeForUpdateSpending(spendingUpdate.GroupId, userId, oldDebts, newDebts);
+			await Context.SaveChangesAsync();
+		}
 
-            await OptimizedService.OptimizeForUpdateSpending(spendingUpdate.GroupId, userId, oldDebts, newDebts);
-            await Context.SaveChangesAsync();
-        }
+		public async Task DeleteSpending(long userId, DeleteSpending spendingData, long groupId)
+		{
+
+			var currentSpending = await Context.Spendings
+											   .Include(x => x.Debtors)
+											  .SingleOrDefaultAsync(x => x.Id == spendingData.Id);
+
+			if (currentSpending == null)
+				throw new ResourceGoneException("spending");
+
+			if (currentSpending.CreditorUserId != userId)
+				throw new ResourceForbiddenException("not_creditor");
+
+			var affectedUsers = new HashSet<long>() { userId };
```

Comment:
Nem látom, hogy hol kerülnek be a spendingben felvett emberek az affectedusers set-be

Reply:
Igen, eddig csak a creditor volt benne. Betettem őket.